### PR TITLE
Rapid7 - Threat Command (IntSights) pack release 3.1.0

### PR DESCRIPTION
**Pull request by crestdatasystems**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description

- Added the network_type parameter for fetch incidents to filter on the network type of alert.
- Added the retry mechanism for 429 and 5xx errors.
- Updated incident fetching mechanism to exclude attachment retrieval during polling. Instead, a new playbook has been introduced specifically for handling attachment retrieval. This implementation is expected to significantly enhance the overall performance of the incident fetching mechanism.
- Updated the help README for setting up the integration instance.
- Updated the test module to include more checks on the integration parameters.
- Added the loggers at the required places.
- Updated the threat-command-alert-csv-get command to return CSV data in the form of JSON.
- Added the tooltip for multiple integration parameters.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/IntSight/ReleaseNotes/3_1_0.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
+#### Incident Fields
+
+- **Rapid7 ThreatCommand CSV**
+  - Added the more columns.
```
Comment (adi88d) [inline Packs/IntSight/ReleaseNotes/3_1_0.md:5]:
```suggestion
- **Rapid7 ThreatCommand CSV**
  - Added more columns.
```

File: Packs/IntSight/ReleaseNotes/3_1_0.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+#### Incident Types
+
+- **Rapid7 ThreatCommand Alert**
+  - Updated the incident type to be attached to the playbook.
```
Comment (adi88d) [inline Packs/IntSight/ReleaseNotes/3_1_0.md:10]:
```suggestion
  - Updated the default playbook for the incident type.
```

File: Packs/IntSight/Integrations/Rapid7ThreatCommand/Rapid7ThreatCommand.py [hunk 3]
```diff
@@ -563,11 +565,10 @@ def alert_fetch_parser(
                 ],
                 "is_closed": dict_safe_get(obj, ["Closed", "IsClosed"]),
                 "sub_type": dict_safe_get(obj, ["Details", "SubType"]),
-                "attachments": attachments,
             }
         )
 
-    def parse_incident(self, alert: dict, attachments: List[dict[str, Any]]) -> dict:
+    def parse_incident(self, alert: dict) -> dict:
```
Comment (adi88d) [inline Packs/IntSight/Integrations/Rapid7ThreatCommand/Rapid7ThreatCommand.py:571]: just want to verify with you,
the incidents are now fetched without attachments and csv files from the integration, instead they will added later with `Retrieve Alert Attachments - Rapid7 ThreatCommand` playbook inside the incident?

**Review by adi88d — CHANGES_REQUESTED** [review]
Great Job!
Thanks a lot for this contribution, It is highly appreciated.
see my comments below

File: Packs/IntSight/Integrations/Rapid7ThreatCommand/Rapid7ThreatCommand.py (hunk 3 above)
Comment (crestdatasystems) [reply Packs/IntSight/Integrations/Rapid7ThreatCommand/Rapid7ThreatCommand.py:571]: Yes, correct.

**Comment by melamedbn** [comment]
Hi @crestdatasystems,

Great work on the contribution. Just one small adjustment that could help our users:

Please rename the 'ThreatCommand Layout' tab to a meaningful pretty name so the user would have an idea what he's going to see there.

Best regards,
Ben

**Review by melamedbn — APPROVED** [review]

**Comment by crestdatasystems** [comment]
@adi88d @melamedbn Thanks for reviewing this PR. We've incorporated all the suggested changes please let us know if you need any additional details.

**Comment by adi88d** [comment]
Hi @crestdatasystems , the code looks good!

We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.

**Pull request merged**